src/common/mailing.ts

```diff
@@ -171,3 +179,134 @@ export const addPrivateSourceJoinParams = ({
 
   return urlObj.toString();
 };
+
+const ITEMS_PER_DESTROY = 4000;
+const ITEMS_PER_IDENTIFY = 250;
+
+interface SyncSubscriptionsWithActiveStateProps {
+  con: DataSource;
+  users: GetUsersActiveState;
+}
+
+export const syncSubscriptionsWithActiveState = async ({
+  con,
+  users: { inactiveUsers, downgradeUsers, reactivateUsers },
+}: SyncSubscriptionsWithActiveStateProps) => {
+  await blockingBatchRunner({
+    data: reactivateUsers,
+    runner: async (batch) => {
+      const validReactivateUsers = await con.getRepository(User).find({
+        select: ['id'],
+        where: { id: In(batch), cioRegistered: false },
+      });
+
+      if (validReactivateUsers.length === 0) {
+        return true;
+      }
+
+      await blockingBatchRunner({
```

Comment:
Why do we need batch runner inside a batch runner? Let's do one batch size (the smaller among the two) and keep the flow simple

Reply:
Yeah, we can do that. This is just so we will do the fetch only once.